staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -235,6 +235,16 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string.
+	// If both Message and MessageExpression are present on a Rule, then MessageExpression will be used if validation
+	// fails. If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
+	// as if the MessageExpression field were unset.
```

Comment:
we don't allow `message` to contain line-breaks. do we have the same restriction for values produced from messageExpression (either by forbidding them or by stripping them out)? if so, document and test for that

Reply:
Right, `messageExpression` output is not allowed to have line breaks. This is tested, but you're right that it's not documented - I'll update the docs.

Reply:
(marking as resolved, docs now mention that line breaks are not allowed in the output)